### DISCUSSION TITLE
How to calculate spectral fluence of charged particle in egs_chamber

### DISCUSSION THREAD

**Discussion by YapingQi** [nrc-cnrc/EGSnrc#1066] [General]
I am modeling an ion chamber in egs_chamber and able to score the deposited dose.  I'd like to know how to score the electron fluence and track length in the air cavity region? Thank you in advance for any suggestions.

**Comment by rtownson** [comment]
For volumetric fluence scoring in egs++, see below. This is a relatively new feature:

https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/classEGS__VolumetricFluence.html

**Reply by YapingQi** [reply]
Thank you for this information! I tried to add egs_fluence_scoring in ausgab object section. I run the script in egs_chamber, but got nothing about fluence results. Can you provide more details?
[try_fluence_script.txt](https://github.com/nrc-cnrc/EGSnrc/files/12925183/try_fluence_script.txt)

**Reply by rtownson** [reply]
There were several typos. Here are some fixes. Make sure to set the volumes correctly.

```
:start ausgab object definition:
    #:start ausgab object:
        library = egs_track_scoring
        name = tracks
        #stop scoring = 1000
    :stop ausgab object:
#     :start ausgab object:
        library = egs_dose_scoring
        name = doseScoring
        medium dose = yes
        dose regions = chamber_cavity_label
        volume = 0.57732
    :stop ausgab object:
    :start ausgab object:
        name    = id-string            # Arbitrary identifying string
        library = egs_fluence_scoring  # Library name
        type    = volumetric           # Score in a volume
        scoring particle = electron# photon, or electron, or positron
        #source particle  = photon, or electron, or positron# Optional. Only required to score primary fluence.
        score primaries = no     # Defaults to `no`.
        score spectrum  = yes    # Defaults to `no`, define energy grid
        #verbose         = no    # Defaults to `no`.
        normalization   = 1         # User-requested normalization. Defaults to 1.
        :start energy grid:# Default: 128 linear energy bins between 1 keV and 1 MeV
            number of bins = 128
            minimum kinetic energy = 0.001
            maximum kinetic energy = 1
            scale = linear #linear(default) or logarithmic
        :stop energy grid:
        :start volumetric scoring:# Enter as many as scoring regions.
            scoring regions = chamber_cavity_label
            volumes = 1
            method  = flurz # flurz or stpwr (default)or stpwrO5 # For charged particle scoring.
        :stop volumetric scoring:
    :stop ausgab object:

:stop ausgab object definition:
```

**Comment by YapingQi** [comment]
Thank you! One more question, I ran the script and the following errors appeared. It seems that I should re-install EGSnrc with the newest version. Is there other way to update the egs_fluence_scoring feature rather than uninstall & install the whole software. 
=========================================================================EGS_Library::load(): failed to load library /root/es/EGSnrc/HEN_HOUSE/egs++/dso/linux/libegs_fluence_scoring.so
  error was: /root/es/EGSnrc/HEN_HOUSE/egs++/dso/linux/libegs_fluence_scoring.so: cannot open shared object file: No such file or directory
EGS_ObjectFactory::createObject(): failed to load the library egs_fluence_scoring from /root/es/EGSnrc/HEN_HOUSE/egs++/dso/linux
EGS_ObjectFactory::createObjects(): 1 errors occured while creating objects

**Reply by rtownson** [reply]
Right, this feature is only in the 2023 release. Note that while it is possible to update your current installation, it can be tricky so is not recommended. The safest way is to do a fresh installation (keep your old one as a backup), and then copy your old input files into the new egs_home that it creates. 

https://github.com/nrc-cnrc/EGSnrc/releases/tag/v2023a
